go/vt/sqlparser/sql.y

```diff
@@ -6868,6 +6868,26 @@ FOR UPDATE
   {
     $$ = ForUpdateLock
   }
+| FOR UPDATE NOWAIT
```

Comment:
not sure if there is a "better" way to do this, but i saw this type of pattern elsewhere so seemed to make sense.